admin/nodes/views.py

```diff
@@ -497,6 +499,72 @@ def get_context_data(self, **kwargs):
         context['resource_type'] = 'node'
         return context
 
+class NodeModifyStorageUsage(TemplateView):
+    template_name = 'nodes/modify_storage_caps.html'
+
+    def get_object(self, queryset=None):
+        node = Node.load(self.kwargs.get('guid')) or Registration.load(self.kwargs.get('guid'))
+        return {
+            'id': node.id,
+        }
+
+    def get_context_data(self, **kwargs):
+        context = super(NodeModifyStorageUsage, self).get_context_data(**kwargs)
+        node = Node.load(self.kwargs.get('guid')) or Registration.load(self.kwargs.get('guid'))
+        context['id'] = node._id
+        context['public_cap'] = node.custom_storage_usage_limit_public or STORAGE_LIMIT_PUBLIC
+        context['private_cap'] = node.custom_storage_usage_limit_private or STORAGE_LIMIT_PRIVATE
+        return context
+
+    def post(self, request, *args, **kwargs):
+        node = Node.load(self.kwargs.get('guid')) or Registration.load(self.kwargs.get('guid'))
+        new_private_cap = request.POST.get('private-cap-input')
+        new_public_cap = request.POST.get('public-cap-input')
+        if float(new_private_cap) != (node.custom_storage_usage_limit_private or STORAGE_LIMIT_PRIVATE):
+            node.custom_storage_usage_limit_private = new_private_cap
+
+        if float(new_public_cap) != (node.custom_storage_usage_limit_public or STORAGE_LIMIT_PUBLIC):
+            node.custom_storage_usage_limit_public = new_public_cap
+
+        node.save()
+        return redirect(reverse_node(self.kwargs.get('guid')))
+
+class NodeRecalculateStorage(NodeDeleteBase):
+    template_name = 'nodes/recalculate_node_storage.html'
+
+    def get_object(self, queryset=None):
+        return Node.load(self.kwargs.get('guid')) or Registration.load(self.kwargs.get('guid'))
+
+    def delete(self, request, *args, **kwargs):
+        node = self.get_object()
+        update_storage_usage_cache(node.id, node._id)
+        if isinstance(node, (Node, Registration)):
+            return redirect(reverse_node(self.kwargs.get('guid')))
+
+    def get_context_data(self, **kwargs):
+        context = super(NodeRecalculateStorage, self).get_context_data(**kwargs)
+        context['link'] = 'nodes:recalculate-node-storage'
+        context['resource_type'] = 'node'
+        return context
+
+class NodeMakePrivate(NodeDeleteBase):
+    template_name = 'nodes/make_private.html'
+
+    def get_object(self, queryset=None):
+        return Node.load(self.kwargs.get('guid')) or Registration.load(self.kwargs.get('guid'))
+
+    def delete(self, request, *args, **kwargs):
+        node = self.get_object()
+        # TODO: update to force set private
```

Comment:
What does this mean? Why can't you force private?

Reply:
I wasn't sure if we were going to throw an error within `node.set_privacy()` when the node is over the private limit.